### DISCUSSION TITLE
How to deploy a turborepo?

### DISCUSSION THREAD

**Discussion by rikurainio** [coollabsio/coolify#2716] [Q&A]
Hey

To start, I am writing to tell people about my experience. maybe it helps someone.

I have deployed a Turborepo that has two apps: (next, socketio) and they share a same database package
I used docker compose and manually built my custom images for the apps.

I probably could have added build commands with build args to my docker compose command, but my 2GB Hetzner VPS made the image builds extremely slow if I had Coolify on, so every time I wanted to generate docker images, I would have to stop my Coolify containers.

I let Coolify adjust my docker-compose and add Traefik to it. Adding a domain for my next app was simple.

Also, I would be interested in examples on the best practices when it comes to deploying a Turborepo monorepo to self-hosted Coolify.

**Comment by ImBIOS** [comment]
https://github.com/coollabsio/coolify-examples/tree/main/turbo-nextjs

read the [README.md](https://github.com/coollabsio/coolify-examples/blob/main/turbo-nextjs/README.md)

> # Coolify Configuration 
> 
> 
> ## Server build (NodeJS)
> - Set `Build Pack` to `nixpacks`.
> - Set the start command to `cd apps/<app-name> && npm run start`.
>    - `npm` could be different depending on the package manager you are using (`Nixpacks` decides based on the lock file).
> - That's all.
